test/e2e/virtual/workspaces/virtual_workspace_test.go

```diff
@@ -146,6 +152,136 @@ func TestWorkspacesVirtualWorkspaces(t *testing.T) {
+					return
+				}
+				assert.Equal(t, testData.workspace1.Name, workspace1.Name)
+				defer server.Artifact(t, func() (runtime.Object, error) {
+					return server.kcpClient.TenancyV1alpha1().Workspaces().Get(ctx, testData.workspace1.Name, metav1.GetOptions{})
+				})
+				workspaceURL := ""
+				if err := server.kcpExpect(workspace1, func(w *tenancyv1alpha1.Workspace) error {
+					if !conditions.IsTrue(w, tenancyv1alpha1.WorkspaceURLValid) {
+						return fmt.Errorf("Workspace %s is not valid: %s", w.Name, conditions.GetMessage(w, tenancyv1alpha1.WorkspaceURLValid))
+					}
+					workspaceURL = w.Status.BaseURL
+					return nil
+				}); err != nil {
+					t.Errorf("did not see the workspace created and valid in KCP: %v", err)
+					return
+				}
+
+				if err := server.virtualWorkspaceExpectations[0](func(w *tenancyv1alpha1.WorkspaceList) error {
+					if len(w.Items) != 1 || w.Items[0].Name != workspace1.Name {
+						return fmt.Errorf("expected only one workspace (%s), got %#v", workspace1.Name, w)
+					}
+					return nil
+				}); err != nil {
+					t.Errorf("did not see the workspace created in personal virtual workspace: %v", err)
+					return
+				}
+
+				req := vwUser1Client.TenancyV1alpha1().RESTClient().Get().Resource("workspaces").Name(workspace1.Name).SubResource("kubeconfig").Do(ctx)
+				if req.Error() != nil {
+					t.Errorf("error retrieving the kubeconfig for workspace %s: %v", workspace1.Name, err)
+				}
+				kcpConfigCurrentContextName := kcpServerKubeconfig.CurrentContext
+				kcpConfigCurrentContext := kcpServerKubeconfig.Contexts[kcpConfigCurrentContextName]
+				if kcpConfigCurrentContext == nil {
+					t.Errorf("kcp Kubeconfig is invalid:\n %+v", kcpServerKubeconfig)
+					return
+				}
+				kcpConfigCurrentCluster := kcpServerKubeconfig.Clusters[kcpConfigCurrentContext.Cluster]
+				if kcpConfigCurrentCluster == nil {
+					t.Errorf("kcp Kubeconfig is invalid:\n %+v", kcpServerKubeconfig)
+					return
+				}
+
+				expectedKubeconfigCluster := kcpConfigCurrentCluster.DeepCopy()
+				expectedKubeconfigCluster.Server = workspaceURL
+				expectedKubeconfig := &api.Config{
+					CurrentContext: "personal/" + workspace1.Name,
+					Contexts: map[string]*api.Context{
+						"personal/" + workspace1.Name: {
+							Cluster: "personal/" + workspace1.Name,
+						},
+					},
+					Clusters: map[string]*api.Cluster{
+						"personal/" + workspace1.Name: expectedKubeconfigCluster,
+					},
+				}
+				expectedKubeconfigContent, err := clientcmd.Write(*expectedKubeconfig)
+				if err != nil {
+					t.Errorf("error writing the content of the expected kubeconfig for workspace %s: %v", workspace1.Name, err)
+					return
+				}
+				workspaceKubeconfigContent, err := req.Raw()
+				if err != nil {
+					t.Errorf("error retrieving the content of the kubeconfig for workspace %s: %v", workspace1.Name, err)
+					return
+				}
+
+				assert.YAMLEq(t, string(expectedKubeconfigContent), string(workspaceKubeconfigContent))
```

Comment:
We have file comparison helpers - do we want this to be in a golden file, so it's easy to read/diff/update as needed?

Reply:
Alternatively, do we want to use semantic equality here, since the ordering of the lists in the structure is not important?